07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,258 @@
+        System.out.println("StoreMap in text representation : " + mapAsString);
+        System.out.println();
+
+        /**
+         * Construct a single StoreList and StoreMap by parser from the provided String
+         */
+
+        StoreList convertedList = StoreList.listOf(listAsString);
+        StoreMap convertedMap = StoreMap.mapOf(mapAsString);
+
+        System.out.println("StoreList parsed from String is : " + convertedList);
+        System.out.println("StoreMap parsed from String is : " + convertedMap);
+        System.out.println();
+
+        /**
+         * Creating a parser for a specific string and retrieving values of the Map.
+         *
+         */
+        ComplexDataParser parser = ComplexData.parser(mapAsString);
+        for (StoreMap p = parser.nextMap(); p != null; p = parser.nextMap()) {
+            Optional<String> name = p.find(".name").strings().first();
+            Optional<Integer> age = p.find(".age").integers().first();
+
+            // Similarly we can retrieve other values from StoreList and StoreMap.
+            System.out.println("Name retrieved by Complex data parser is : " + name.get());
+            System.out.println("Age retrieved by Complex data parser is : " + age.get());
+        }
+
+        /**
+         * Find operaions
+         * find(spec) Selection querying for efficient selecting of arbitrary data inside CDTs by type, position, and key, in various combinations
+         * get(), first(), last(), sole(), isPresent(), size()
+         */
+
+        Record<String> record = ComplexRecord.mutableRecord("rec", StoreMap.newMap().with("map", map)).toRecord();
+        System.out.println("Age is : " + Record.find(".map.age").get(0).apply(record).get().value());
+        System.out.println("Get first age : " + Record.find(".map.age").first().apply(record).get().value());
+        System.out.println("Get last age : " + Record.find(".map.age").last().apply(record).get().value());
+        System.out.println("Get sole age : " + Record.find(".map.age").sole().apply(record).value());
+        System.out.println("Get age from end : " + Record.find(".map.age").get(-1).apply(record).get().value());
+        System.out.println("5th Age is present ? : " + Record.find(".map.age").get(5).apply(record).isPresent());
+        System.out.println("Size of map is : " + Record.find(".map.[]").apply(record).size());
+
+        Record<String> rec2 = ComplexRecord.mutableRecord("foofoo", map).toRecord();
+        System.out.println("First value in map is : " + Record.find(".[]").first().apply(rec2).get().value());
+        System.out.println();
+
+        StoreList newList = StoreList.newList().with(10, 12, 14, 16);
+        Record<String> rec3 = ComplexRecord.mutableRecord("foo", StoreMap.newMap().with("list", newList)).toRecord();
+        System.out.println("Value in list at 2nd index is 14 ? : " + Record.find(".list.[]").get(2).is(typedValue(14)).test(rec3));
```

Comment:
If someone had a single record why would they test it in this functional way? This seems weird, and a little contrived.